File: Object-Oriented-Programming/Inheritance/inheritance.java
// Java Implementation of Inheritance in OOP

class Person {
    // Base class representing a generic person
    protected String name;
    protected int age;
    
    public Person(String name, int age) {
        this.name = name;
        this.age = age;
    }
    
    public void introduce() {
        System.out.println("Hello, my name is " + name + " and I am " + age + " years old.");
    }
}

// Derived Class
class Teacher extends Person {
    private String subject;
    
    public Teacher(String name, int age, String subject) {
        super(name, age);
        this.subject = subject;
    }
    
    public void teach() {
        System.out.println("I am teaching " + subject + ".");
    }
}

// Example Usage
public class Main {
    public static void main(String[] args) {
        Teacher t1 = new Teacher("Alice", 35, "Mathematics");
        t1.introduce();
        t1.teach();
    }
}

File: Object-Oriented-Programming/Inheritance/inheritance.py
# Python Implementation of Inheritance in OOP

class Person:
    """Base class representing a generic person."""
    def __init__(self, name, age):
        self.name = name
        self.age = age
    
    def introduce(self):
        print(f"Hello, my name is {self.name} and I am {self.age} years old.")

# Derived Class
class Teacher(Person):
    """Derived class representing a teacher, inheriting from Person."""
    def __init__(self, name, age, subject):
        super().__init__(name, age)
        self.subject = subject
    
    def teach(self):
        print(f"I am teaching {self.subject}.")

# Example Usage
t1 = Teacher("Alice", 35, "Mathematics")
t1.introduce()
t1.teach()
